api/clients/eigenda_client_v2.go

```diff
@@ -0,0 +1,142 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/hashicorp/go-multierror"
+	"math/rand"
+)
+
+// EigenDAClientV2 provides the ability to get blobs from the relay subsystem, and to send new blobs to the disperser.
+type EigenDAClientV2 interface {
+	// GetBlob iteratively attempts to retrieve a given blob with key blobKey from the relays listed in the blobCertificate.
+	GetBlob(ctx context.Context, blobKey corev2.BlobKey, blobCertificate corev2.BlobCertificate) ([]byte, error)
+	GetCodec() codecs.BlobCodec
+	Close() error
+}
+
+type eigenDAClientV2 struct {
+	clientConfig EigenDAClientConfig
```

Comment:
Where is EigenDAClientConfig located? Is it defined in the V1 client?
Feel like we should redefine it here even if its the same (for now)?
Otherwise it makes me wonder why it is the same.

Reply:
also think this can just be called `config`

Reply:
[6848663](https://github.com/Layr-Labs/eigenda/pull/972/commits/684866345456cc7bc8fdab20ba62b1e5b05e0382)

I created a new v2 client config, only including necessary values for now. I will add additional values as further functionality is added to the v2 client

Reply:
sg! Will resolve this convo, but let's follow up in https://github.com/Layr-Labs/eigenda/pull/972#discussion_r1878735827